Shopify/polaris-react
# [new DL][Skeleton Page] Ensure Skeleton Page title is the same font size as title rendered in Page component

### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-react/issues/3414

When the new design language is enabled, the page title shrinks when the Skeleton Page component is replaced by the Page component.

![Kapture 2020-10-07 at 14 08 22](https://user-images.githubusercontent.com/1591075/95376546-44b5db80-08af-11eb-965e-f9ceb182e035.gif)

The font size for the page title should be consistent between the Skeleton Page component and the Page component.

### WHAT is this pull request doing?
This PR implements a solution similar to the one introduced by https://github.com/Shopify/polaris-react/pull/3468. Instead of using JavaScript to conditionally render small or large Display Text when the navigation is collapsed, the Skeleton Page component now relies purely on CSS to set different font sizes based on specific breakpoints.

|New design language|Current design language (no regressions)
|---|---
|![page-title-skeleton-after](https://user-images.githubusercontent.com/8534230/95606337-30491e80-0a28-11eb-9d31-cf49f0e16ef8.gif)|![page-title-skeleton-prev-design-lang](https://user-images.githubusercontent.com/8534230/95606352-350dd280-0a28-11eb-992b-f7abe5ae8246.gif)

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

#### Testing in Storybook
1. Run `dev s` in the Polaris React repo to open Storybook
2. Look at the Skeleton Page examples and ensure there are no regressions

|Page title defined|Page title undefined
|---|---
|![image](https://user-images.githubusercontent.com/8534230/96026381-6ae5f900-0e24-11eb-9284-8bf608d1b83f.png)|![image](https://user-images.githubusercontent.com/8534230/96026379-68839f00-0e24-11eb-97a6-75a94d9ce206.png)


#### Testing against a Shopify Web
1. Check out this branch and configure Shopify Web to consume your local Polaris React repo
2. Visit a page that uses the Skeleton Page component (e.g., /admin/apps, /admin/products, /admin/dashboards)

|Viewport width|Page title defined|Page title undefined|
|---|---|---
|Mobile|![skeleton-page-title-defined-mobile](https://user-images.githubusercontent.com/8534230/96028009-c2856400-0e26-11eb-8cfd-685ae696b5b2.gif)|![skeleton-page-title-null-mobile](https://user-images.githubusercontent.com/8534230/96027996-be594680-0e26-11eb-97c2-0f92ad63db27.gif)
|Desktop|![skeleton-page-title-defined-wide](https://user-images.githubusercontent.com/8534230/96028056-d03ae980-0e26-11eb-9b80-8e7213a7bf05.gif)|![skeleton-page-title-null-wide](https://user-images.githubusercontent.com/8534230/96028071-d4670700-0e26-11eb-8dac-9fe8d1c61008.gif)

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit
